x/vesting/keeper/permissioned_decorator.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/vesting/types"
+)
+
+// PermissionedVestingDecorator prevents invalid msg types from being executed
+type PermissionedVestingDecorator struct {
+	vestingKeeper       Keeper
+	disabledMsgTypeURLs []string
+}
+
+func NewPermissionedVestingDecorator(vestingKeeper Keeper, msgTypeURLs []string) PermissionedVestingDecorator {
+	return PermissionedVestingDecorator{
+		vestingKeeper:       vestingKeeper,
+		disabledMsgTypeURLs: msgTypeURLs,
+	}
+}
+
+// AnteHandle rejects vesting messages that signer does not have permissions
+// to create vesting account.
+func (pvd PermissionedVestingDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	for _, msg := range tx.GetMsgs() {
+		typeURL := sdk.MsgTypeURL(msg)
+		for _, disabledTypeURL := range pvd.disabledMsgTypeURLs {
+			if typeURL == disabledTypeURL {
+				// Check if vesting tx signer is 1
+				if len(msg.GetSigners()) != 1 {
```

Comment:
why are we blocking multisigs?

Reply:
As I see, all vesting msgs on [cosmos-sdk](https://github.com/cosmos/cosmos-sdk/blob/d6d929843bbd331b885467475bcb3050788e30ca/x/auth/vesting/types/msgs.go#L76) or [evmos](https://github.com/dymensionxyz/evmos/blob/4ab8372bced4a8509de9abc4c0ad53b50e34b1b3/x/vesting/types/msg.go#L121) only contains 1 signer. WDYT

Reply:
ok we can start with that